src/models/blueprint-cloner.js

```diff
@@ -0,0 +1,84 @@
+import path from 'path';
+import { copySync } from 'fs-extra';
+import walkSync from 'walk-sync';
+
+import { fileExists } from '../util/fs';
+
+const FILES_BLACKLIST = ['.ds_store', '.git', '.gitkeep'];
+
+export default class BlueprintCloner {
+  constructor(blueprint, options) {
+    this.blueprint = blueprint;
+    this.options = options || {};
+    this.ui = options.ui;
+  }
+
+  clone() {
+    const blueprint  = this.blueprint;
+    this.ui.writeInfo('cloning blueprint...');
+
+    const cloneToDirectory = this.cloneToDirectory();
```

Comment:
The directory we clone to will be determined at the settings level, something like `rc.get('path.bpTarget`). So by this point in execution, you'll either have a falsy value or a validated, existing path you can write to.  Falsy values should generate a warning or an error.  

It seems to me that making assumptions based on the search path order, is risking unintended consequences.  Better to get 'permission' with an explicity defined path.

Reply:
I totally agree, it is risky to make assumptions based on the search path order and it does make sense to get that `permission` from the explicit path.

I was aware that I will need to change the `cloneDirectory` to something like `rc.get('path.bpTarget')`. 

I will make the change once the new `rc` api is ready.